code/game/objects/items/holy_weapons.dm

```diff
@@ -322,19 +322,25 @@
 	w_class = WEIGHT_CLASS_HUGE
 	force = 5
 	slot_flags = ITEM_SLOT_BACK
-	block_chance = 50
-	var/shield_icon = "shield-red"
+	var/user_shield_icon = "shield-red"
 
-/obj/item/nullrod/staff/worn_overlays(isinhands)
-	. = list()
-	if(isinhands)
-		. += mutable_appearance('icons/effects/effects.dmi', shield_icon, MOB_SHIELD_LAYER)
+/obj/item/nullrod/staff/Initialize()
+	. = ..()
+	AddComponent(/datum/component/shielded, max_charges = 1, recharge_start_delay = 10 SECONDS, shield_icon = user_shield_icon, shield_inhand = TRUE, run_hit_callback = CALLBACK(src, .proc/shield_damaged))
```

Comment:
you need to hold this in your hand for it to work, and not in your back or suit storage slots, yes?